src/metabase/analyze/fingerprint/insights.clj

```diff
@@ -66,22 +68,23 @@
              idx (.nextInt rng c+1)]
          (aset counter 0 c+1)
          (cond
-           (< c n)   (aset reservoir c x)
-           (< idx n) (aset reservoir idx x)))))))
+           (< c n)   (aset reservoir c (f x))
+           (< idx n) (aset reservoir idx (f x))))))))
 
 (defn- simple-linear-regression
   "Faster and more efficient implementation of `kixi.stats.estimate/simple-linear-regression`. Computes some of squares
-  on each step, and on the completing step returns `[offset slope]`."
-  [fx fy]
+  on each step, and on the completing step returns `[offset slope]`. Additionally accepts `x-link-fn` and `y-link-fn`
+  functions which should be double->double."
+  [fx, fy, ^clojure.lang.IFn$DD x-link-fn, ^clojure.lang.IFn$DD y-link-fn]
```

Comment:
TBH, it's the first time I do it too 😄. Maybe, it's worth rewriting the regression function to completely lose the generality and just perform `math/log` in `if` branch. I'm not sure if there ever be changes and additions in `trendline-function-families`. If not, it may make sense to cement this behavior.

Reply:
I personally find the `x-link-fn` names quite undescriptive and something tricky for future readers to figure out. Will branching not hurt performance a bit too? It could be nice to go fully declarative and have say an `x-scale` argument that could be `:linear` or `:log`, a nice extension point if we need any fancier kernels in future.

Reply:
Branching only matters in some very specific low-level code, certainly wouldn't matter here. Besides, virtual dispatch is almost always slower than branching.

Alright, let me implement your suggestions:).

Reply:
Damn, I forgot to disable auto-merge after you approved. I'll submit in another PR.